### ISSUE TITLE
Disappearing from the map

### ISSUE THREAD

**Issue by ETLegacyTeam** [etlegacy/etlegacy#633]
Author: @saukko28 (Saukko)
Date: 2014-11-19
Redmine Issue: 633, https://dev.etlegacy.com/issues/633

---

I found a couple of places where I could reproduce this. I took a demo in Fueldump to show what this bug is and screenshots in Railgun too.
The demo shows, what happens when you go in front of the tank. You can't see anything and it will only go away when you \kill yourself.
The screenshots were taken before I ever tried Fueldump. This happened at the specfic spot. First I was in front of the carriage. At once when I jumped on top of it, my screen went grey, and when I moved backwards I could get the grey screen off. The third screenshot is to show from the enemy point of view what is happening. But if I just had demo you could see the guy disappearing and then appearing all over again.

I will try to provide a demo of the Railgun case.

-*S

---

Related commit(s):
26be6249e55363467ca978d43f7ba527e61c8427 - mod: 'TransposeMatrix swapped arguments' commit re..

**Comment by ETLegacyTeam** [comment]
From: @rafal1137 (RaFaL)
Date: 2014-11-19T23:25:52Z

---

I tested it and in my case mostly I get crushed by tank I even made a demo

**Comment by ETLegacyTeam** [comment]
From: @saukko28 (Saukko)
Date: 2014-11-20T01:51:46Z

---

I tried out Capuzzo and also Goldrush, same issue. Going in front of a tank makes me disappear. It seems to have something to do with movers... I made two demos. I actually didn't have this current issue when I created this ticket: http://dev.etlegacy.com/issues/628 So something between the time I created 628 and 633 have messed something up.

-*S

**Comment by ETLegacyTeam** [comment]
From: @IR4T4 (IR4T4)
Date: 2014-11-20T08:29:09Z

---

I've had the same issue while specing a bot but I don't know how exactly to reproduce.

Related to movers:
Saukko or RaFal, what happens if you revert this commit? http://dev.etlegacy.com/projects/etlegacy/repository/revisions/96c159456880b58b939c40c21d90f6970fdd33b2/diff/src/cgame/cg_ents.c 

In case this is our bug we should also have a look at http://dev.etlegacy.com/projects/etlegacy/repository/revisions/96c159456880b58b939c40c21d90f6970fdd33b2/diff/src/cgame/cg_ents.c

Edit: I did revert both and the fuledump tank issue is gone.

**Comment by ETLegacyTeam** [comment]
From: @rmarquis (Spyhawk)
Date: 2014-11-20T08:46:56Z

---

> 96c15945

This was my first thought yesterday night when reading this bug report. What is weird is that this commit is actually the "correct" one (in an academic sense), so there might be some other intentional bug introduced somewhere to "counterfix" this one.

Edit: Confirmed, this is the responsible commit. Now reverted by IR4.
Edit2: DIsregard the above comment. I'm not sure why I saw some errors while the TransposeMatrix code was cleaned, since I cannot find anything relevant anymore. My bad :/
